app/javascript/components/city-name.jsx

```diff
@@ -0,0 +1,36 @@
+import React from 'react';
+import { Query } from 'react-apollo';
+import gql from 'graphql-tag';
+import { WhiteTitleLarge } from '@components/typography';
+import { LoadingMessage, ErrorMessage } from '@messages/default-messages';
+
+const getCityName = gql`
+  {
+    primaryLocation {
+      cityName
+    }
+  }
+`;
+
+export const City = () => (
+  <Query query={getCityName}>
+    {({ loading, error, data }) => {
+      if (loading) {
+        return <LoadingMessage />;
+      }
+
+      if (error) {
+        return <ErrorMessage />;
+      }
+
+      const { cityName } = data.primaryLocation;
+      if (!cityName) {
+        return null;
+      }
+
+      return <WhiteTitleLarge>{cityName}</WhiteTitleLarge>;
```

Comment:
```suggestion
      return (cityName && <WhiteTitleLarge>{cityName}</WhiteTitleLarge>);
```

Nitpick: There is nothing wrong with the early return as written. I do, however, see `&&` and `||` used heavily for control flow in JSX. In my opinion the one-liner version is slightly more idiomatic.

Reply:
Ah yes, makes perfect sense! I agree with you, its much neater too.